R/SEIaImIsRD.R

```diff
@@ -0,0 +1,352 @@
+#' @aliases initial_conditions<-,SEIaImIsRD
+#' @export
+#'
+setMethod(
+  "initial_conditions<-",
+  "SEIaImIsRD",
+  function(object,
+           value = list(S = NA_real_, E = NA_real_,
+           I_asymptomatic = NA_real_, I_mild = NA_real_, I_severe = NA_real_,
+           R = NA_real_,
+           D = NA_real_)) {
+    init_pop_list <- value
+    names(init_pop_list) <- object@initial_condition_names
+    object@initial_conditions <- init_pop_list
+    # check if values are valid
+    errors <- character()
+    # check whether all required initial population groups are set
+    is_na_pop <- is.na(init_pop_list)
+    if (sum(is_na_pop) != 0) {
+      msg <- paste("Missing initial setting for population group:",
+                   paste(names(init_pop_list)[is_na_pop], collapse = ", ")
+      )
+      errors <- c(errors, msg)
+    }else{
+      # check whether the sum of initial population is normalized to 1
+      sum_init_pop <- sum(unlist(init_pop_list))
+      if (sum_init_pop != 1) {
+        msg <- "Sum of initial population is not 1, please normalize"
+        errors <- c(errors, msg)
+      }
+    }
+    if (length(errors) == 0) {
+      object@initial_conditions <- init_pop_list
+      object
+      }
+    else stop(paste(errors, ", please check and rerun transmission_parameters<-.\n"))
+  })
+
+
+#' @describeIn SEIaImIsRD Setter method for transmission parameters
+#' of the SEIaImIsRD model.
+#'
+#' @param object An object of class SEIaImIsRD
+#' @param value A numeric named list of values for transmission parameters:
+#' beta - a named list of the effective contact rate from each infected group (i.e. rate at which an infected individual exposes susceptible), each element with value in [0,1]
+#' kappa - rate of progression from exposed to infectious (the reciprocal is the incubation period),
+#' omega - rate at which recovered individuals become susceptible,
+#' p_symptom - a named list of fraction of different infected groups, only need to specify values for mild and severe symptom groups, whose sum should be <=1. 
+#' gamma - a list of the rate of removal of each infected group (i.e. recovery rate of an infected individual), each element with value in [0,1]
+#' mu - a list of the rate of disease-caused mortality of each infected group, each element with value in [0,1]
+#' @return An object of class SEIaImIsRD with initial population and parameters
+#' @aliases transmission_parameters<-,SEIaImIsRD-method
+#' @export
+#'
+setMethod(
+  "transmission_parameters<-",
+  "SEIaImIsRD",
+  function(
+    object,
+    value = list(beta = list(i_asymptomatic = NA_real_, i_mild = NA_real_, i_severe = NA_real_), 
+    kappa = NA_real_, 
+    omega = NA_real_,
+    p_symptom = list(i_mild = NA_real_, i_severe = NA_real_), 
+    gamma = list(i_asymptomatic = NA_real_, i_mild = NA_real_, i_severe = NA_real_), 
+    mu = list(i_asymptomatic = NA_real_, i_mild = NA_real_, i_severe = NA_real_))) {
```

Comment:
Similar to the comment about notation below, can we just make the setters: beta = list(asymptomatic=, mild=, severe=)? In other words, drop the "i_" in all cases.